src/app/dataset/depublication/depublication.component.ts

```diff
@@ -310,11 +355,14 @@ export class DepublicationComponent implements OnDestroy {
           }
         });
       })
-    ) as Observable<MoreResults<RecordPublicationInfo>>;
+    ) as Observable<MoreResults<RecordDepublicationInfo>>;
 
     this.depublicationSubscription = polledDepublicationData.subscribe(
       ({ results, more }) => {
-        this.depublicationData = results;
+        this.depublicationData = results.map((entry: RecordDepublicationInfoDeletable) => {
+          entry.deletion = this.depublicationDeletions.indexOf(entry.recordId) > -1;
+          return entry as RecordDepublicationInfoDeletable;
```

Comment:
According to my IDE also here the assertion is not necessary and the type of the expression is not changed.